modules/images/image-loading-optimization/detection.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Detection for image loading optimization.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Prints the script for detecting loaded images and the LCP element.
+ *
+ * @todo This should eventually only print the script if metrics are needed.
+ * @todo This script should not be printed if the page was requested with non-removal (non-canonical) query args.
+ */
+function ilo_print_detection_script() {
+
+	// TODO: Also abort if we don't need any new page metrics due to the sample size being full.
+	if ( ilo_is_page_metric_storage_locked() ) {
+		return;
+	}
+
+	$serve_time = ceil( microtime( true ) * 1000 );
+
+	/**
+	 * Filters the time window between serve time and run time in which loading detection is allowed to run.
+	 *
+	 * Allow this amount of milliseconds between when the page was first generated (and perhaps cached) and when the
+	 * detect function on the page is allowed to perform its detection logic and submit the request to store the results.
+	 * This avoids situations in which there is missing detection metrics in which case a site with page caching which
+	 * also has a lot of traffic could result in a cache stampede.
+	 *
+	 * @since n.e.x.t
+	 * @todo The value should probably be something like the 99th percentile of Time To Last Byte (TTLB) for WordPress sites in CrUX.
+	 *
+	 * @param int $detection_time_window Detection time window in milliseconds.
+	 */
+	$detection_time_window = apply_filters( 'perflab_image_loading_detection_time_window', 5000 );
+
+	$detect_args = array(
+		$serve_time,
+		$detection_time_window,
+		WP_DEBUG,
+		rest_url( ILO_REST_API_NAMESPACE . ILO_PAGE_METRIC_STORAGE_ROUTE ),
+		wp_create_nonce( 'wp_rest' ),
+	);
+	wp_print_inline_script_tag(
+		sprintf(
+			'import detect from %s; detect( ...%s )',
+			wp_json_encode( add_query_arg( 'ver', PERFLAB_VERSION, plugin_dir_url( __FILE__ ) . 'detection/detect.js' ) ),
```

Comment:
This should use the constant specific to this module, as `PERFLAB_VERSION` may not be defined (as this is going to be a standalone plugin eventually).

Reply:
OK, so should the `IMAGE_LOADING_OPTIMIZATION_VERSION` constant then not use the `PERFLAB_VERSION` either? And then should `IMAGE_LOADING_OPTIMIZATION_VERSION` be used here?

Reply:
@westonruter `PERFLAB_VERSION` should only be referenced in `IMAGE_LOADING_OPTIMIZATION_VERSION` actually. It will be replaced by a script with the relevant standalone version when creating the standalone plugin build.

Reply:
So this should be modified to just use `IMAGE_LOADING_OPTIMIZATION_VERSION`?

```suggestion
			wp_json_encode( add_query_arg( 'ver', IMAGE_LOADING_OPTIMIZATION_VERSION, plugin_dir_url( __FILE__ ) . 'detection/detect.js' ) ),
```

Reply:
@westonruter Yes, correct.

Reply:
Done in 19a2c3b